webapi/src/main/scala/org/knora/webapi/messages/v1/responder/groupmessages/GroupMessagesV1.scala

```diff
@@ -0,0 +1,288 @@
+  */
+case class UpdateGroupApiRequestV1(propertyIri: String,
+                                     newValue: String) {
+    def toJsValue = GroupV1JsonProtocol.updateGroupApiRequestV1Format.write(this)
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Messages
+
+/**
+  * An abstract trait representing a request message that can be sent to [[GroupsResponderV1]].
+  */
+sealed trait GroupsResponderRequestV1 extends KnoraRequestV1
+
+// Requests
+/**
+  * Get all information about all groups.
+  *
+  * @param infoType is the type of the group information: full or short.
+  * @param userProfile the profile of the user making the request.
+  */
+case class GroupsGetRequestV1(infoType: GroupInfoType, userProfile: Option[UserProfileV1]) extends GroupsResponderRequestV1
+
+
+/**
+  * Get everything about a single group identified through it's IRI.
+  *
+  * @param iri Iri of the group.
+  * @param infoType is the type of the group information: full or short.
+  * @param userProfileV1 the profile of the user making the request.
+  */
+case class GroupInfoByIRIGetRequest(iri: IRI, infoType: GroupInfoType.Value, userProfileV1: Option[UserProfileV1]) extends GroupsResponderRequestV1
+
+
+/**
+  * Find everything about a single group identified through it's shortname. Because it is only required to have unique
+  * names inside a project, it is required to supply the name of the project.
+  *
+  * @param projectIri the IRI of the project the group is part of.
+  * @param groupName the name of the group.
+  * @param infoType is the type of the project information.
+  * @param userProfileV1 the profile of the user making the request.
+  */
+case class GroupInfoByNameGetRequest(projectIri: IRI, groupName: String, infoType: GroupInfoType.Value, userProfileV1: Option[UserProfileV1]) extends GroupsResponderRequestV1
+
+
+/**
+  * Requests the creation of a new group.
+  *
+  * @param newGroupInfo the [[NewGroupInfoV1]] information for creating the new group.
+  * @param userProfile the user profile of the user creating the new group.
+  * @param apiRequestID the ID of the API request.
+  */
+case class GroupCreateRequestV1(newGroupInfo: NewGroupInfoV1,
+                                userProfile: UserProfileV1,
+                                apiRequestID: UUID) extends GroupsResponderRequestV1
+
+/**
+  * Request updating of an existing group.
+  *
+  * @param groupIri the IRI of the group to be updated.
+  * @param propertyIri the IRI of the property to be updated.
+  * @param newValue the new value for the property.
+  * @param userProfile the user profile of the user requesting the update.
+  * @param apiRequestID the ID of the API request.
+  */
+case class GroupInfoUpdateRequestV1(groupIri: webapi.IRI,
+                                    propertyIri: webapi.IRI,
+                                    newValue: Any,
```

Comment:
Please refactor this so it doesn't use `Any`. It's error-prone because it disables type checking, and it makes the code harder to understand, because it's not clear which types are actually possible.

You could do this by adding an `Option` member variable to `GroupInfoUpdateRequestV1` for each property that can be updated.